src/modules/mavlink/streams/OPEN_DRONE_ID_SYSTEM.hpp

```diff
@@ -0,0 +1,93 @@
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#ifndef OPEN_DRONE_ID_SYSTEM_HPP
+#define OPEN_DRONE_ID_SYSTEM_HPP
+
+#include <uORB/topics/home_position.h>
+
+class MavlinkStreamOpenDroneIdSystem : public MavlinkStream
+{
+public:
+	static MavlinkStream *new_instance(Mavlink *mavlink) { return new MavlinkStreamOpenDroneIdSystem(mavlink); }
+
+	static constexpr const char *get_name_static() { return "OPEN_DRONE_ID_SYSTEM"; }
+	static constexpr uint16_t get_id_static() { return MAVLINK_MSG_ID_OPEN_DRONE_ID_SYSTEM; }
+
+	const char *get_name() const override { return get_name_static(); }
+	uint16_t get_id() override { return get_id_static(); }
+
+	unsigned get_size() override
+	{
+		return _home_position_sub.advertised() ? MAVLINK_MSG_ID_OPEN_DRONE_ID_SYSTEM_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES : 0;
+	}
+
+private:
+	explicit MavlinkStreamOpenDroneIdSystem(Mavlink *mavlink) : MavlinkStream(mavlink) {}
+
+	uORB::Subscription _home_position_sub{ORB_ID(home_position)};
+
+	bool send() override
+	{
+		home_position_s home_position;
+
+		if (_home_position_sub.update(&home_position)) {
+			if (home_position.valid_alt && home_position.valid_hpos) {
+				mavlink_open_drone_id_system_t msg{};
+				msg.target_component = 0; // 0 for broadcast
+				msg.target_system = 0; // 0 for broadcast
+				// msg.id_or_mac // Only used for drone ID data received from other UAs.
+				msg.operator_location_type = MAV_ODID_OPERATOR_LOCATION_TYPE_TAKEOFF;
+				msg.classification_type = MAV_ODID_CLASSIFICATION_TYPE_UNDECLARED;
+				msg.operator_latitude = home_position.lat * 1e7;
+				msg.operator_longitude = home_position.lon * 1e7;
+				msg.area_count = 1;
+				msg.area_radius = 0;
+				msg.area_ceiling = -1000;
+				msg.area_floor = -1000;
+				msg.category_eu = MAV_ODID_CATEGORY_EU_UNDECLARED;
+				msg.class_eu = MAV_ODID_CLASS_EU_UNDECLARED;
+				msg.operator_altitude_geo = home_position.alt;
+				msg.timestamp = home_position.timestamp; // TODO: 32 bit Unix Timestamp in seconds since 00:00:00 01/01/2019.
```

Comment:
@BluemarkInnovations we spoke about this on the dev call. It makes most sense to plug a gps into your computer (or use one internal if on a tablet) and stream from QGC back up. It may or may not take a QGC change but it can already read NMEA and live position if you have gps. 

That seems like the best path and then as @dagar said, if we have rtk and related in use we can start to benefit system wide from that data. 

Let me know your thoughts.

Reply:
> st sense to plug a gps into your computer (or use one internal if on a tablet) and stream from QGC back up. It may or may not take a QGC change but it can already read NMEA and live position if you have gps.

I do know that there are customers using ArduPilot and a Herelink running QGC. In that product there is RemoteID functionality already. Not sure if that is part of QGC or something Hex implemented. So it seems to me a good approach. And if the RemoteID functionality is already be part of QGC, only a few steps are needed to get a dynamic location in the RemoteID signal.